src/highdicom/sr/templates.py

```diff
@@ -584,6 +892,92 @@ def __init__(
             )
             self.append(role_in_procedure_item)
 
+    @property
+    def name(self) -> str:
+        """str: name of the person"""
+        return self[0].value
+
+    @property
+    def login_name(self) -> Union[str, None]:
+        """Union[str, None]: login name of the person"""
+        matches = [
+            item for item in self
+            if item.name == codes.DCM.PersonObserverLoginName
+        ]
+        if len(matches) > 0:
+            return matches[0].value
+        return None
+
+    @property
+    def organization_name(self) -> Union[str, None]:
+        """Union[str, None]: name of the person's organization"""
+        matches = [
+            item for item in self
+            if item.name == codes.DCM.PersonObserverOrganizationName
+        ]
+        if len(matches) > 0:
+            return matches[0].value
+        return None
+
+    @property
+    def role_in_organization(self) -> Union[str, None]:
+        """Union[str, None]: role of the person in the organization"""
+        matches = [
+            item for item in self
+            if item.name == codes.DCM.PersonObserverRoleInTheOrganization
+        ]
+        if len(matches) > 0:
+            return matches[0].value
+        return None
+
+    @property
+    def role_in_procedure(self) -> Union[str, None]:
+        """Union[str, None]: role of the person in the procedure"""
+        matches = [
+            item for item in self
+            if item.name == codes.DCM.PersonObserverRoleInThisProcedure
+        ]
+        if len(matches) > 0:
+            return matches[0].value
+        return None
+
+    @classmethod
+    def from_sequence(
+        cls,
+        sequence: Sequence[Dataset]
+    ) -> 'PersonObserverIdentifyingAttributes':
+        """Construct instance from a sequence of datasets.
+
+        Parameters
+        ----------
+        sequence: Sequence[pydicom.dataset.Dataset]
+            Datasets representing SR Content Items of template
+            TID 1003 "Person Observer Identifying Attributes"
+
+        Returns
+        -------
+        highdicom.sr.templates.PersonObserverIdentifyingAttributes
+            Content Sequence containing SR Content Items
+
+        """
+        attr_codes = [
+            ('name', codes.DCM.PersonObserverName),
+            ('login_name', codes.DCM.PersonObserverLoginName),
+            ('organization_name',
+             codes.DCM.PersonObserverOrganizationName),
+            ('role_in_organization',
+             codes.DCM.PersonObserverRoleInTheOrganization),
+            ('role_in_procedure',
+             codes.DCM.PersonObserverRoleInThisProcedure),
+        ]
+        kwargs = {}
+        for dataset in sequence:
+            content_item = ContentItem.from_dataset(dataset)
+            for param, name in attr_codes:
+                if content_item.name == name:
+                    kwargs[param] = content_item.value
+        return PersonObserverIdentifyingAttributes(**kwargs)
```

Comment:
I have accepted all of your suggestions, but just realized that this may actually a really bad idea. `cls` may not be the class for which the `classmethod` has been defined, but a derived class. Here, we really want to construct an object of the given type not an instance of a derived type. I think we should undo all of these changes

Reply:
Realized that you were right about this.